### PR TITLE
feat: refactor linter into smaller modules

### PR DESCRIPTION
**Pull request by nichmor**
Checklist
* [ ] Added a ``news`` entry
* [ ] Regenerated schema JSON if schema altered (`python conda_smithy/schema.py`)

Aims to make #1970 to make easier to review and implement by splitting the linter into smaller modules.

### CODE REVIEW

**Comment by nichmor** [comment]
hey @beckermr ! This PR aims to split linter into smaller modules ( like, `linter.py,` `lints.py `and `hints.py `)

**Review by beckermr — CHANGES_REQUESTED** [review]
Let's break this into smaller PRs please. I'd suggest each one be about a net change of 500 lines. You could 

- make one pr that makes the new module structure and utils
- make several prs that refactor the lints 
- make several prs that refactor the hints

As it is now, this pr is too big for me to review effectively. You may be able to find someone else to review it of course, in which case you can ignore me.

**Comment by wolfv** [comment]
@beckermr I am currently sitting next to @nichmor and would be happy to do a deep walkthrough through the PR with him. If you are OK, I would merge it after my review.

**Comment by beckermr** [comment]
Sure go for it.

File: conda_smithy/lint_recipe.py [hunk 1]
```diff
@@ -181,10 +103,14 @@ def lintify_meta_yaml(
     # If the recipe_dir exists (no guarantee within this function) , we can
     # find the meta.yaml within it.
     meta_fname = os.path.join(recipe_dir or "", "meta.yaml")
+    # if not os.path.exists(meta_fname):
```
Comment (wolfv) [inline conda_smithy/lint_recipe.py:106]: remove, or uncomment?

File: conda_smithy/linter/lints.py [hunk 2]
```diff
@@ -0,0 +1,797 @@
...
+import re
+
+import fnmatch
+import io
+import itertools
+import os
+import sys
+
+
+if sys.version_info[:2] < (3, 11):
```
Comment (wolfv) [inline conda_smithy/linter/lints.py:11]: what's this about? :)

**Comment by wolfv** [comment]
Two small comments, rest LGTM :)

**Comment by jaimergp** [comment]
Don't forget to include the `news/` file in the next ones (you can put the one for this PR later) 🙏

**Comment by nichmor** [comment]
> Don't forget to include the `news/` file in the next ones (you can put the one for this PR later) 🙏

my fault! 
Added it here: https://github.com/conda-forge/conda-smithy/pull/1982

**Pull request merged**